volume/service/service_test.go

```diff
@@ -141,7 +143,7 @@ func TestServiceGet(t *testing.T) {
 
 	v, err = service.Get(ctx, "test")
 	assert.Assert(t, err)
-	assert.Assert(t, is.DeepEqual(created, v))
+	assert.Assert(t, is.DeepEqual(created, v, cmpopts.IgnoreFields(types.Volume{}, "CreatedAt")))
```

Comment:
Why would these have different created time?

Reply:
Wonder if it's due to `t.Parallel()`; could it be that we're inspecting a volume named `test` from another test?

Reply:
It may be because for testing it uses the FakeVolume type which has the method:
https://github.com/moby/moby/blob/5037c5a8ce762b46638378b7a7d1081572eadba1/volume/testutils/testutils.go#L72

Which is used to assign a value to the CreatedAt field.

Reply:
Ah, yes, that definitely looks like the problem. We should probably set an actual `created` date when creating the volume;

https://github.com/moby/moby/blob/5037c5a8ce762b46638378b7a7d1081572eadba1/volume/testutils/testutils.go#L93-L100


And update `NewFakeVolume ()` to take that date as an argument

https://github.com/moby/moby/blob/5037c5a8ce762b46638378b7a7d1081572eadba1/volume/testutils/testutils.go#L40-L49

Reply:
We probably don't need the date argument. Just take the current time inside `NewFakeVolume`, store it and always use the same date later.

Reply:
Ah, yes, that should be sufficient